protocol/strict-derivation.md

```diff
@@ -0,0 +1,269 @@
+frame. The strict ordering rules, as detailed above, would in such a case drop any pre-existing
+staging channel, and the batch queue should always be empty right after the pipeline origin
+progresses (open question: is this true, in view of lingering `undecided` batches?).
+
+# Steady Batch Derivation
+
+Steady Batch Derivation changes how invalid batches and payload attributes at different stages in
+the derivation pipieline are treated.
+- Batch queue: if a batch is found to be invalid at the batch queue stage (`BatchDrop`), it is immediately replaced by an empty
+  batch, leading to deposit-only (first batch of an epoch) or empty payload attributes.
+- Engine queue: if the payload attributes are found by the EL client to be
+  [`INVALID`](https://github.com/ethereum/go-ethereum/blob/7fd7c1f7dd9ba8d90399df2f080e4101ae37a255/beacon/engine/errors.go#L63)
+  they are replaced by deposit-only/empty payload attributes and reinserted into the EL.
+
+At this point, there are no open questions about the design of this protocol change. I assume that
+the actual implementation will raise questions on certain details.
+
+# Activation rules & actions
+
+I propose to activate the new rules by L1 inclusion block timestamp, which is most convenient and clean
+for derivation protocol upgrades. It's accessible in most parts of the derivation pipeline as the
+stage's _origin_ (not to be confused with the _L1 origin of an L2 block_).
+Note that this is in contrast to how Delta activated span batches, which was by first parsing the
+span batch and then checking the span batch's L1 origin timestamp.
+
+However, we need to specify how existing buffered frames, channels and batches are treated at
+activation, since the stricter rules of Holocene take effect.
+To start with a derivation pipeline state that is consistent with Holocene rules, I propose the
+following special activation effects to take place when the pipeline's origin reaches the activation
+timestamp, which is similar to applying Holocene rules retroactively.
+- Channel Bank:
+  1. All channels that don't consist of contiguous frames that include the first, are dropped. 
+  2. All but the youngest (in terms of L1 inclusion) channel are dropped.
+  3. This optional left-over channel
+     will be the start staging channel for the Holocene channel bank.
+
+The principle of eager derivation of later stages should guarantee at this point that all batches that could have
+been applied directly on top of the current safe chain have already been exhausted from the batch
+queue. So what is left should be `undecided` and `future` batches. (Note: I'm not sure on this one.)
+I see two options on how to handle a non-empty batch queue at this point:
+- Option 1: Drop future batches, continue resolving undecided batches, if any are left, and apply
+new Holocene rules.
+- Option 2: The Batch Queue will just start applying Holocene rules from this moment onwards. This will then
```

Comment:
Option 3: When the L1 origin reaches the Holocene activation block, discard all batches in the batch queue.  Could also be applied to the Channel Bank to give us a nice clean starting point.

op-batcher would have to be aware of that rule and consider any blocks it submitted in channels that didn't close prior to the holocene activation block as needing to be resubmitted.

Reply:
I also thought about this option, but deemed it too drastic. Hearing that you support it makes me consider it! I like its simplicity.

We just need to be careful that the batcher doesn't end up in a very unlikely situation where we're batching with calldata and have a single block that needs to be sent over two calldata frame txs that span across the Holocene activation.

Reply:
Thanks for flagging, I also thought about this option, but deemed it too drastic. Hearing that you support this option, I certainly consider it! I like the simplicity of it. 
We just need to be careful with the batcher to arrive in a very unlikely situation where we're sending calldata txs and need to batch a single block that doesn't fit into a single calldata tx and then have that channel span across the Holocene activation. Very unlikely, but possible.

Reply:
Even if we did find ourselves in that situation, discarding that in-progress submission would work right? It would need to be retried after the threshold.

Or, we could follow Adrian's suggestion and drop all batches, but do it some time *before* holocene activation, holding a ban on batches until holocene passes. This would allow for the unlikely big-block to resolve prior to passing the threshold.

Reply:
Yes I think it is reasonable to implement some sort of safety behavior for the batcher close to the Holocene activation. We didn't do a good job of adding special hardfork activation logic to the batcher in the past but I think past batcher operating experience has shown that it may be worthwhile adding some.

Reply:
Made this the new and preferred Option 1.